src/main/java/uk/gov/hmcts/ccd/data/caseaccess/AMCaseUserRepository.java

```diff
@@ -1,39 +1,99 @@
 package uk.gov.hmcts.ccd.data.caseaccess;
 
-import com.google.common.collect.Lists;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.google.common.collect.ImmutableSet;
 import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.reform.amlib.AccessManagementService;
+import uk.gov.hmcts.reform.amlib.enums.AccessorType;
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessGrant;
+import uk.gov.hmcts.reform.amlib.models.ExplicitAccessMetadata;
+import uk.gov.hmcts.reform.amlib.models.ResourceDefinition;
+import uk.gov.hmcts.reform.amlib.models.UserCaseRolesEnvelope;
+import uk.gov.hmcts.reform.amlib.models.UserCasesEnvelope;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
-import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
+import javax.sql.DataSource;
+import javax.transaction.Transactional;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.amlib.enums.Permission.READ;
 
 @Named
 @Singleton
-@Qualifier(AMCaseUserRepository.QUALIFIER)
+@Qualifier(AMCaseUserRepository.ACCESS_MANAGEMENT_QUALIFIER)
 public class AMCaseUserRepository implements CaseUserRepository {
 
-    public static final String QUALIFIER = "am";
+    public static final String ACCESS_MANAGEMENT_QUALIFIER = "accessManagement";
+    private static final String CASE_CONSTANT = "case";
+
+    private AccessManagementService accessManagementService;
 
-    @PersistenceContext
-    private EntityManager em;
+    public AMCaseUserRepository(@Qualifier("amDataSource") DataSource dataSource) {
+        accessManagementService = new AccessManagementService(dataSource);
+    }
 
     @Override
+    @Transactional
     public void grantAccess(String jurisdictionId, String caseTypeId, String caseReference, Long caseId, String userId, String caseRole) {
+        ResourceDefinition resourceDefinition =
+            //TODO: What should be the resourceType and resourceName.
```

Comment:
Why adding a TODO here? Excuse me if I am not up to date here. Are we missing this information from the ticket?

Reply:
Keeping it to resourcetype for now. Will improvise if change is needed in future.